src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -0,0 +1,211 @@
+            _position = position;
+            _metadataAsSourceFileService = metadataAsSourceFileService;
+
+            CancellationToken = cancellationToken;
+        }
+
+        public override Task OnCompletedAsync()
+        {
+            var results = Array.Empty<VSReferenceItem>();
+            lock (_gate)
+            {
+                if (_resultsChunk.IsEmpty())
+                {
+                    return Task.CompletedTask;
+                }
+
+                results = _resultsChunk.ToArray();
+                _resultsChunk.Clear();
+            }
+
+            _progress.Report(results);
+            return Task.CompletedTask;
+        }
+
+        public async override Task OnDefinitionFoundAsync(DefinitionItem definition)
+        {
+            if (!_definitionToId.ContainsKey(definition))
+            {
+                // Assinging a new id to the definition
+                _definitionToId.Add(definition, _id);
+
+                // Creating a new VSReferenceItem for the definition
+                var definitionItem = await GenerateVSReferenceItemAsync(
+                    _id, definitionId: _id, _document, _position, definition.SourceSpans.FirstOrDefault(),
+                    definition.DisplayableProperties, _metadataAsSourceFileService, definition.GetClassifiedText(),
+                    symbolUsageInfo: null, CancellationToken).ConfigureAwait(false);
+
+                if (definitionItem.Location != null)
+                {
+                    AddAndReportResultsIfAtMax(definitionItem);
+                    _id++;
+                }
+            }
+        }
+
+        public async override Task OnReferenceFoundAsync(SourceReferenceItem reference)
+        {
+            if (_definitionToId.TryGetValue(reference.Definition, out var definitionId))
+            {
+                // Creating a new VSReferenceItem for the reference
+                var referenceItem = await GenerateVSReferenceItemAsync(
+                    _id, definitionId, _document, _position, reference.SourceSpan,
+                    reference.AdditionalProperties, _metadataAsSourceFileService, definitionText: null,
+                    reference.SymbolUsageInfo, CancellationToken).ConfigureAwait(false);
+
+                if (referenceItem.Location != null)
+                {
+                    AddAndReportResultsIfAtMax(referenceItem);
+                    _id++;
+                }
+            }
+        }
+
+        private void AddAndReportResultsIfAtMax(VSReferenceItem item)
+        {
+            var results = Array.Empty<VSReferenceItem>();
+            lock (_gate)
+            {
+                _resultsChunk.Add(item);
+                if (_resultsChunk.Count < MaxResultsChunkSize)
+                {
+                    return;
+                }
+
+                results = _resultsChunk.ToArray();
+                _resultsChunk.Clear();
+            }
+
+            _progress.Report(results);
```

Comment:
what is the return tyep of .Report?

Reply:
It looks like a void return type: https://docs.microsoft.com/en-us/dotnet/api/system.iprogress-1.report?view=netframework-4.8#System_IProgress_1_Report__0_